lib/net/vnc.rb

```diff
@@ -288,7 +336,7 @@ def read_packet type
         len = socket.read(4).unpack('N')[0]
         @mutex.synchronize { @clipboard = socket.read len }
       else
-        raise NotImplementedError, 'unhandled server packet type - %d' % type
+        @fb.handle_response type
```

Comment:
I am not sure I like this. The framebuffer supports 4 modes 0-3:

```
0 - handle_fb_updated
1 - handle_colormap_update
2 - bell
3 - handle_server_cuttext
```

And we are already handling 3 ourselves, and I do not know what bell is, so I don't know if we care, but it is nice that it is handled so our client don't blow up in case it gets a bell.

We could change this code to:

```ruby
    def read_packet type
      case type
      when 0..2
        @fb.handle_response type
      when 3 # ServerCutText
        socket.read 3 # discard padding bytes
        len = socket.read(4).unpack('N')[0]
        @mutex.synchronize { @clipboard = socket.read len }
      else
        raise NotImplementedError, 'unhandled server packet type - %d' % type
      end
    end
```

But just reading the code makes it still a bit magic, so maybe we could add proper comments with vocabulary from https://tools.ietf.org/html/rfc6143#section-7.6

```ruby
    def read_packet type
      case type
      when 0..2
        # 0 - FramebufferUpdate
        # 1 - SetColorMapEntries
        # 2 - Bell
        @fb.handle_response type
      when 3 # ServerCutText
        socket.read 3 # discard padding bytes
        len = socket.read(4).unpack('N')[0]
        @mutex.synchronize { @clipboard = socket.read len }
      else
        raise NotImplementedError, 'unhandled server packet type - %d' % type
      end
    end
```

Or very repetitive (but clearer):

```ruby
    def read_packet type
      case type
      when 0 # FramebufferUpdate
        @fb.handle_response type
      when 1 # SetColorMapEntries
        @fb.handle_response type
      when 2 # Bell
        @fb.handle_response type
      when 3 # ServerCutText
        socket.read 3 # discard padding bytes
        len = socket.read(4).unpack('N')[0]
        @mutex.synchronize { @clipboard = socket.read len }
      else
        raise NotImplementedError, 'unhandled server packet type - %d' % type
      end
    end
```